core/CliMulti/RequestParser.php

```diff
@@ -62,9 +64,13 @@ private function filterNonArchivingJobs($commands)
 
     private function getPsLinesWithCliMulti(string $psOutput)
     {
+        $instanceId = SettingsPiwik::getPiwikInstanceId();
         $lines = explode("\n", $psOutput);
         $lines = array_map('trim', $lines);
-        $lines = array_filter($lines, function ($line) {
+        $lines = array_filter($lines, function ($line) use ($instanceId) {
+            if (!empty($instanceId) && strpos($line, 'matomo-domain=' . $instanceId) === false) {
+                return false;
+            }
```

Comment:
not sure if this is a problem for cloud, but should we also check if `empty($instanceId) && strpos($line, 'matomo-domain=') !== false`? Will merge regardless after tests run, just thought it might be important.

Reply:
it should be not needed. When an instanceId is configured in the config, then we expect it to be always called with matomo-domain. And if no instanceId is set we assume it doesn't matter.